lib/authentication/auth_web.js

```diff
@@ -2,36 +2,45 @@
  * Copyright (c) 2015-2021 Snowflake Computing Inc. All rights reserved.
  */
 
-var util = require('../util');
-var rest = require('../global_config').rest;
+const util = require('../util');
+const rest = require('../global_config').rest;
 
-var net = require('net');
-var querystring = require('querystring');
-var URLUtil = require('./../../lib/url_util');
+const net = require('net');
+const querystring = require('querystring');
+const URLUtil = require('./../../lib/url_util');
+const Util = require('./../../lib/util');
 
 /**
  * Creates an external browser authenticator.
  *
  * @param {String} host
  * @param {module} webbrowser
  * @param {module} httpclient
+ * @param {module} browserActionTimeout
  *
  * @returns {Object}
  * @constructor
  */
-function auth_web(host, webbrowser, httpclient)
-{
-  var open = typeof webbrowser !== "undefined" ? webbrowser : require('open');
-  var axios = typeof httpclient !== "undefined" ? httpclient : require('axios');
+function auth_web(host, webbrowser, httpclient, browserActionTimeout) {
 
-  var host = host;
-  var port = rest.HTTPS_PORT;
-  var protocol = rest.HTTPS_PROTOCOL;
-  var proofKey;
-  var token;
-  var data;
+  if (!Util.exists(host)) {
+    throw new Error(util.format("Invalid value for host ", host));
+  }
+  if (!Util.number.isPositiveInteger(browserActionTimeout)) {
+    throw new Error(util.format("Invalid value for browser action timeout ", browserActionTimeout));
+  }
 
-  var successResponse = 'HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\nConnection: close\r\n\r\nYour identity was confirmed and propagated to Snowflake Node.js driver. You can close this window now and go back where you started from.';
+  const open = typeof webbrowser !== "undefined" ? webbrowser : require('open');
+  const axios = typeof httpclient !== "undefined" ? httpclient : require('axios');
+
+  const browserTimeout = browserActionTimeout || WAIT_FOR_BROWSER_ACTION_TIMEOUT
```

Comment:
it's checked above that `browserActionTimeout ` is positive integer - we don't need to set up default

Reply:
ok